sql/set.go

```diff
@@ -80,3 +81,43 @@ func (p *planner) getStringVal(name string, values parser.Exprs) (string, error)
 	}
 	return string(s), nil
 }
+
+func (p *planner) SetTimeZone(n *parser.SetTimeZone) (planNode, error) {
+	d, err := p.evalCtx.EvalExpr(n.Value)
+	if err != nil {
+		return nil, err
+	}
+	var offset int64
+	switch v := d.(type) {
+	case parser.DString:
+		location := string(v)
+		if location == "DEFAULT" || location == "LOCAL" {
+			location = "UTC"
+		}
+		if _, err := time.LoadLocation(location); err != nil {
+			return nil, err
+		}
+		p.session.Timezone = &Session_Location{Location: location}
+
+	case parser.DInterval:
+		offset = int64(v.Duration / time.Second)
+
+	case parser.DInt:
+		offset = int64(v) * 60 * 60
+
+	case parser.DFloat:
+		offset = int64(float64(v) * 60.0 * 60.0)
+
+	default:
+		return nil, fmt.Errorf("bad time zone value: %v", n.Value)
+	}
+	if offset != 0 {
+		day := int64(24 * 60 * 60)
+		if offset > day || offset < -day {
+			return nil, fmt.Errorf("time zone offset %v out of range", d)
```

Comment:
but actually, why bother with this restriction? I may have a scenario where i'm loading data with timestamps that omit a timezone and i want to shift them all by two weeks; this would be a convenient way to do so. Postgres does not limit this, and I don't think we should either.